### PR TITLE
refactor(k8s client): update k8s packages based on idiomatic maya

### PR DESCRIPTION
**Pull request by AmitKumarDas**
This commit refactors kubernetes client and deployment packages as
per idiomatic maya standards:
- All lines of code within 80 chars
- All code comments within 80 chars
- Add sync.Once to get singleton client instance
- Refactor code to make it readable

Signed-off-by: AmitKumarDas <amit.das@mayadata.io>

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

File: pkg/kubernetes/client/v1alpha1/client.go [hunk 1]
```diff
@@ -134,6 +135,21 @@ func New(opts ...OptionFn) *Client {
...
+var (
+	instance *Client
+	once     sync.Once
+)
+
+// Instance returns a singleton instance of
+// this client
+func Instance() *Client {
+	once.Do(func() {
+		instance = New()
```
Comment (vishnuitta) [inline pkg/kubernetes/client/v1alpha1/client.go]: this being single instance to exists, should we check for NULL for instance before doing allocation?

File: pkg/kubernetes/client/v1alpha1/client.go [hunk 2]
```diff
@@ -134,6 +135,21 @@ func New(opts ...OptionFn) *Client {
 	return c
 }
 
+var (
+	instance *Client
+	once     sync.Once
```
Comment (vishnuitta) [inline pkg/kubernetes/client/v1alpha1/client.go]: :+1: for using `sync`

File: pkg/kubernetes/client/v1alpha1/client.go (hunk 1 above)
Comment (AmitKumarDas) [reply pkg/kubernetes/client/v1alpha1/client.go]: We are leaving this check to library. I believe library does it.
https://golang.org/src/sync/once.go

Comment (vishnuitta) [reply pkg/kubernetes/client/v1alpha1/client.go]: sounds good :+1:

**Pull request merged**